server/graphql/invoice/src/mutations/outbound_return/insert.rs

```diff
@@ -49,3 +76,70 @@ pub enum InsertErrorInterface {
     OtherPartyNotVisible(OtherPartyNotVisible),
     OtherPartyNotASupplier(OtherPartyNotASupplier),
 }
+
+fn map_error(error: ServiceError) -> Result<InsertErrorInterface> {
+    use StandardGraphqlError::*;
+    let formatted_error = format!("{:#?}", error);
+
+    let graphql_error = match error {
+        ServiceError::OtherPartyNotVisible => {
+            return Ok(InsertErrorInterface::OtherPartyNotVisible(
+                OtherPartyNotVisible,
+            ))
+        }
+        ServiceError::OtherPartyNotASupplier => {
+            return Ok(InsertErrorInterface::OtherPartyNotASupplier(
+                OtherPartyNotASupplier,
+            ))
+        }
+
+        // Standard Graphql Errors
+        ServiceError::InvoiceAlreadyExists => BadUserInput(formatted_error),
+        ServiceError::OtherPartyDoesNotExist => BadUserInput(formatted_error),
+        ServiceError::NewlyCreatedInvoiceDoesNotExist => InternalError(formatted_error),
+        ServiceError::LineInsertError { .. } => InternalError(formatted_error),
+        ServiceError::LineReturnReasonUpdateError { .. } => InternalError(formatted_error),
+        ServiceError::DatabaseError(_) => InternalError(formatted_error),
+    };
+
+    Err(graphql_error.extend())
+}
+
+impl InsertInput {
+    pub fn to_domain(self) -> ServiceInput {
+        let InsertInput {
+            id,
+            supplier_id,
+            outbound_return_lines,
+        }: InsertInput = self;
+
+        ServiceInput {
+            id,
+            other_party_id: supplier_id,
+            outbound_return_lines: outbound_return_lines
+                .into_iter()
+                .map(|line| line.to_domain())
+                .collect(),
+        }
+    }
+}
+
+impl OutboundReturnLineInput {
+    pub fn to_domain(self) -> InsertOutboundReturnLine {
+        let OutboundReturnLineInput {
+            id,
+            stock_line_id,
+            number_of_packs_to_return,
+            reason_id,
+            comment,
+        }: OutboundReturnLineInput = self;
+
+        InsertOutboundReturnLine {
+            id,
+            stock_line_id,
+            number_of_packs: number_of_packs_to_return,
+            reason_id,
+            note: comment,
```

Comment:
I'm questioning this mapping - it will exposed at `note` on the InvoiceLineNode after the OutboundReturn invoice is created... should we just call it `note` everywhere in code, and can call it Comment in the UI if that makes more sense?

Reply:
yeah, for sure, great point !